File: Algorithms_4th_Edition/java/src/1/2/book/TestAccumulator.java
import edu.princeton.cs.algs4.StdRandom;

public class TestAccumulator {

    public static void main(String[] args) {
        int t = Integer.parseInt(args[0]);
        Accumulator a = new Accumulator();
        for (int i = 0; i < t; i++) {
            a.addDataValue(StdRandom.uniformDouble());
        }
        System.out.println(a);
    }
}

File: Algorithms_4th_Edition/c/1/2/book/testAccumulator.c
#include <stdio.h>
#include <stdlib.h>
#include "accumulator.h"
#include "..\..\..\lib\std_random.h"

int main(int argc,char * argv[])
{
    if(argc < 2)
    {
        fprintf(stderr,"Usage: %s integer(cnt)\n",argv[0]);
        exit(EXIT_FAILURE);
    }

    setSeedByTime();
    int t = atoi(argv[1]);
    Accumulator a;
    initAccumulator(&a);
    for(int i = 0; i < t; i++)
        addDataValue(&a,uniformDouble());
    char str[100];
    accumulatorToString(&a,str,100);
    printf("%s\n",str);
    return 0;
}
